Description: Allow systemd interaction
 When started by systemd, we need to allow notify access, and access to
 certain /proc areas.
Origin: upstream, https://gitlab.com/apparmor/apparmor/-/merge_requests/236,
 https://gitlab.com/apparmor/apparmor/-/merge_requests/811
Bug: https://gitlab.com/apparmor/apparmor/-/issues/203
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/apparmor/+bug/1952242
Applied-Upstream: https://gitlab.com/apparmor/apparmor/-/commit/924d4e87adbf02cafa169b3eed7892d0d35a2088,
 https://gitlab.com/apparmor/apparmor/-/commit/ee9e61aad284f4edbebbd7cd0e8d9ac452455958
Last-Update: 2021-11-26
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
Ubuntu notes:
 - upstream missed the smbd profile wrt access the systemd notify socket
 - upstream merged the fix for #203 just in the avahi profile and left the bug
   open pending discussion of adding a more generic fix as an abstraction
---
 profiles/apparmor.d/usr.sbin.smbd |   12 ++++++++++++
 1 file changed, 12 insertions(+)

--- apparmor-4.0.0-beta3.orig/profiles/apparmor.d/usr.sbin.smbd
+++ apparmor-4.0.0-beta3/profiles/apparmor.d/usr.sbin.smbd
@@ -26,6 +26,9 @@
 
   signal send set=term peer=samba-bgqd,
 
+  # when started by systemd
+  ptrace read peer=unconfined,
+
   /etc/mtab r,
   /etc/netgroup r,
   /etc/printcap r,
@@ -35,6 +38,13 @@
   /usr/etc/environment r,
   /usr/etc/security/limits.d/ r,
   /usr/etc/security/limits.d/*.conf  r,
+
+  # https://gitlab.com/apparmor/apparmor/-/issues/203
+  # needed when smbd is started by systemd
+  @{PROC}/1/environ r,
+  @{PROC}/cmdline r,
+  @{PROC}/sys/kernel/osrelease r,
+
   /usr/lib*/samba/vfs/*.so mr,
   /usr/lib*/samba/auth/*.so mr,
   /usr/lib*/samba/charset/*.so mr,
@@ -59,6 +69,8 @@
   @{run}/samba/** rk,
   @{run}/samba/ncalrpc/ rw,
   @{run}/samba/ncalrpc/** rw,
+  # when started by systemd
+  @{run}/systemd/notify w,
   /var/spool/samba/** rw,
 
   @{HOMEDIRS}/** lrwk,
